### ISSUE TITLE
AttributeError: module 'tensorflow' has no attribute 'estimator'

### ISSUE THREAD

**Issue by javadnoorb** [tensorflow/tensorflow#13856]
Hi, for some reason 'estimator' isn't an attribute of 'tensorflow' when I test it on Mac. 

When I try:

`tf.estimator.Estimator(model_fn)`

I get this error 

`AttributeError: module 'tensorflow' has no attribute 'estimator'`

This works fine in Linux though. Upgrading pip/conda/tensorflow did not affect this behavior. Is there a workaround?

-Thanks

### System information
- Mac OS X 10.9.5 (x86_64-apple-darwin13.4.0)
- TensorFlow installed from binary
- TensorFlow version v1.3.0-rc1-3628-g49f9c6f89 1.5.0-dev20171020
- Python 3.6.3

**Comment by javadnoorb** [comment]
This was a persistent error that somehow was resolved (after multiple restarts of terminal). Since I don't know how to reproduce it anymore I think this thread can be closed.

**Comment by VictorZuanazzi** [comment]
Hello,

I am having the same problem:
```

import tensorflow as tf

my_optimizer= tf.train.GradientDescentOptimizer(learning_rate=0.00000001)

my_optimizer = tf.contrib.estimator.clip_gradients_by_norm(my_optimizer, 5.0)
Traceback (most recent call last):

  File "<ipython-input-36-49765c0f1466>", line 1, in <module>
    my_optimizer = tf.contrib.estimator.clip_gradients_by_norm(my_optimizer, 5.0)

AttributeError: module 'tensorflow.contrib' has no attribute 'estimator'
```

But estimator is in tensorflow.contrib:
```

help (tensorflow.contrib)
Help on package tensorflow.contrib in tensorflow:

NAME
    tensorflow.contrib

PACKAGE CONTENTS
    all_reduce (package)
    batching (package)
    boosted_trees (package)
    cluster_resolver (package)
    coder (package)
    decision_trees (package)
    eager (package)
    estimator (package)
    feature_column (package)
    fused_conv (package)
    ...
```

I am having the same problem:

import tensorflow as tf

my_optimizer= tf.train.GradientDescentOptimizer(learning_rate=0.00000001)

my_optimizer = tf.contrib.estimator.clip_gradients_by_norm(my_optimizer, 5.0)
Traceback (most recent call last):

  File "<ipython-input-36-49765c0f1466>", line 1, in <module>
    my_optimizer = tf.contrib.estimator.clip_gradients_by_norm(my_optimizer, 5.0)

AttributeError: module 'tensorflow.contrib' has no attribute 'estimator'
But estimator is in tensorflow.contrib:

```
    help (tensorflow.contrib)
Help on package tensorflow.contrib in tensorflow:

NAME
    tensorflow.contrib

PACKAGE CONTENTS
    all_reduce (package)
    batching (package)
    boosted_trees (package)
    cluster_resolver (package)
    coder (package)
    decision_trees (package)
    eager (package)
    estimator (package)
    feature_column (package)
    fused_conv (package)
in fact this same error happens with any function from tensorflow.contrib.

help (tensorflow.contrib.boosted_trees)
Traceback (most recent call last):

  File "<ipython-input-38-b173108b832b>", line 1, in <module>
    help (tensorflow.contrib.boosted_trees)

AttributeError: module 'tensorflow.contrib' has no attribute 'boosted_trees'

```

I am using tensorflow 1.7.
I am using anaconda, spider, windowns 10 .
I am going through this tutorial, if more details of the code are important: https://colab.research.google.com/notebooks/mlcc/first_steps_with_tensor_flow.ipynb?hl=en#scrollTo=ubhtW-NGU802

Any idea of what is wrong?

**Comment by VictorZuanazzi** [comment]
Problem solved.

There was an environment issue, tensorflow was running in its own environment and not at base, at anaconda. A friend reinstalled a few packages at the base environment and deleted the tensorflow environmnet. I can't really say what he did, we had several trials before figuring out the problem and what to do.

**Comment by javadnoorb** [comment]
Good to hear that.

**Comment by mhlr** [comment]
I am having this now. Any hints on how to get rid of it. I have TF-1.7 from Anaconda and I just installe the current tensorflow-hub. I am running on linux.

**Comment by javadnoorb** [comment]
@mhlr seems like for me and Victor this was an installation/environment issue. I recommend uninstalling and reinstalling tensorflow.

**Comment by mhlr** [comment]
Solved. The problem was between the chair and the keyboard :)

**Comment by javadnoorb** [comment]
Haha, glad it's resolved.

**Comment by harikrishnanrajeev** [comment]
me too faced the same issue. Did the following steps
1) uninstall tensor flow
2) reinstall tensor flow and tensor flow hub
and it worked :-)

**Comment by jlinleung** [comment]
> I didn't encounter this error again. Contrib packages apparently are not very stable. So they may change between versions. My guess is that for me it was an installation issue. Have you tried uninstalling tensorflow and reinstalling it?

hi, javadnoorb, i have tried, but it still do not work for me. i am using python 3.6. any other idea?

**Comment by javadnoorb** [comment]
@jlinleung can you confirm that you don't have multiple versions in different paths? Maybe installing a clean version in a new conda/python environment would resolve it. Alternatively (and this can be dangerous) you might want to VERY CAREFULLY delete any folder in your path that contains 'tensorflow' in the name.

**Comment by sometimescasey** [comment]
FYI for anyone who comes across this in future - `tf.estimator` doesn't seem to exist prior to tensorflow r1.1. (Verify here using the branch dropdown at top left: https://github.com/tensorflow/tensorflow/blob/r1.1/tensorflow/python/estimator/estimator.py)

Users above who resolved by uninstalling and reinstalling Tensorflow probably upgraded to most recent version.

**Comment by DesJYX** [comment]
i've got the same problem : /

**Comment by morvan-s** [comment]
I had the same problem, even with upgrading tensorflow.
However, with 'pip freeze | grep tensorflow' I saw that I had a 'tensorflow-estimator' package that keeps being installed even if I uninstalled tensorflow. After uninstalling it and deleting some related folder in site-packages, and re-installing tensorflow, everything worked.
(I'm not sure that this is the best way to solve this problem, but if you are stuck that could help)

**Comment by ocampesato** [comment]
@morvan-s thanks for your suggestion: I renamed two estimator-related directories and now my code works!

**Comment by dgarciale** [comment]
Faced the same issue with tensorflow 2.3.0 and tensorflow-hub 0.10.0.
I just solved it by uninstalling tensorflow-estimator, tensorflow-hub and tensorflow,
then installed tensorflow and tensorflow-hub.

Now it's working :-)

**Comment by chut89** [comment]
Hi guys! Just want to mention that tensorflow.estimator is no longer available in Tensorflow 2.16+. This guild details the migration from Estimator to Keras APIs https://www.tensorflow.org/guide/migrate/migrating_estimator
